src/uct/sm/mm/mm_posix.c

```diff
@@ -31,11 +31,60 @@ static ucs_config_field_t uct_posix_pd_config_table[] = {
   {NULL}
 };
 
+static ucs_status_t uct_posix_test_mem(size_t length, int shm_fd)
+{
+    int *buf;
+    int i, iters, chunk_size = 256 * 1024;
+    size_t size_to_write, remaining, single_write, total_write = 0;
+
+    buf = (int*) ucs_malloc(length, "write buffer");
+    memset(buf, 0, length);
+
+    if (length > chunk_size) {
+        iters = length / chunk_size;
+        size_to_write = chunk_size;
+        remaining = length % chunk_size;
+    } else {
+        iters = 1;
+        size_to_write = length;
+        remaining = 0;
+    }
+
+    for (i = 0; i < iters; i++) {
+        single_write = write(shm_fd, buf, size_to_write);
+        if (single_write <= 0) {
+            ucs_error("Failed to write %zu bytes. %m", size_to_write);
+        } else {
+            total_write += single_write;
+        }
+
+    }
+    if (remaining) {
+        single_write = write(shm_fd, buf, remaining);
+        if (single_write <= 0) {
+            ucs_error("Failed to write remaining %zu bytes. %m", remaining);
+        } else {
+            total_write += single_write;
+        }
+    }
+
+    ucs_free(buf);
+
+    if (length != total_write) {
```

Comment:
you can check for every write that it returned the expected value, and fail immediately.